### ISSUE TITLE
TypeError: ShortNotes\PostType\Note\get_reply_to_markup(): Argument #1 ($post) must be of type WP_Post, null given, called in /var/www/html/wp-content/plugins/shortnotes/includes/post-type-note.php on line 383

### ISSUE THREAD

**Issue by aaronjorbin** [jeremyfelt/shortnotes#44]
This seems to happen on oembed requests. For example, `https://aaron.jorb.in/wp-json/oembed/1.0/embed?url=https%3A%2F%2Faaron.jorb.in%2Ftransparent-salaries-in-wordpress%2F` which currently triggers a fatal error. 

Full stack trace:
```
TypeError: ShortNotes\PostType\Note\get_reply_to_markup(): Argument #1 ($post) must be of type WP_Post, null given, called in /var/www/html/wp-content/plugins/shortnotes/includes/post-type-note.php on line 383
(/var/www/html/wp-content/plugins/shortnotes/includes/post-type-note.php:315)

ShortNotes\PostType\Note\get_reply_to_markup(/var/www/html/wp-content/plugins/shortnotes/includes/post-type-note.php:383)

ShortNotes\PostType\Note\prepend_reply_to_markup(/var/www/html/wp-includes/class-wp-hook.php:324)

WP_Hook::apply_filters(/var/www/html/wp-includes/plugin.php:205)

apply_filters(/var/www/html/wp-includes/formatting.php:3992)

wp_trim_excerpt(/var/www/html/wp-includes/class-wp-hook.php:324)

WP_Hook::apply_filters(/var/www/html/wp-includes/plugin.php:205)

apply_filters(/var/www/html/wp-includes/post-template.php:434)

get_the_excerpt(/var/www/html/wp-content/plugins/wordpress-seo/src/helpers/post-helper.php:84)

Yoast\WP\SEO\Helpers\Post_Helper::get_the_excerpt(/var/www/html/wp-content/plugins/wordpress-seo/src/presentations/indexable-post-type-presentation.php:196)

Yoast\WP\SEO\Presentations\Indexable_Post_Type_Presentation::generate_open_graph_description(/var/www/html/wp-content/plugins/wordpress-seo/src/presentations/abstract-presentation.php:66)

Yoast\WP\SEO\Presentations\Abstract_Presentation::__get(/var/www/html/wp-content/plugins/wordpress-seo/src/presenters/open-graph/description-presenter.php:33)

Yoast\WP\SEO\Presenters\Open_Graph\Description_Presenter::get(/var/www/html/wp-content/plugins/wordpress-seo/src/surfaces/values/meta.php:205)

Yoast\WP\SEO\Surfaces\Values\Meta::__get(/var/www/html/wp-content/plugins/wordpress-seo/src/integrations/front-end/open-graph-oembed.php:116)

Yoast\WP\SEO\Integrations\Front_End\Open_Graph_OEmbed::set_description(/var/www/html/wp-content/plugins/wordpress-seo/src/integrations/front-end/open-graph-oembed.php:94)

Yoast\WP\SEO\Integrations\Front_End\Open_Graph_OEmbed::set_oembed_data(/var/www/html/wp-includes/class-wp-hook.php:326)

WP_Hook::apply_filters(/var/www/html/wp-includes/plugin.php:205)

apply_filters(/var/www/html/wp-includes/embed.php:609)

get_oembed_response_data(/var/www/html/wp-includes/class-wp-oembed-controller.php:132)

WP_oEmbed_Controller::get_item(/var/www/html/wp-includes/rest-api/class-wp-rest-server.php:1193)

WP_REST_Server::respond_to_request(/var/www/html/wp-includes/rest-api/class-wp-rest-server.php:1041)

WP_REST_Server::dispatch(/var/www/html/wp-includes/rest-api/class-wp-rest-server.php:431)

WP_REST_Server::serve_request(/var/www/html/wp-includes/rest-api.php:424)

rest_api_loaded(/var/www/html/wp-includes/class-wp-hook.php:324)

WP_Hook::apply_filters(/var/www/html/wp-includes/class-wp-hook.php:348)

WP_Hook::do_action(/var/www/html/wp-includes/plugin.php:565)

do_action_ref_array(/var/www/html/wp-includes/class-wp.php:418)

WP::parse_request(/var/www/html/wp-includes/class-wp.php:813)

WP::main(/var/www/html/wp-includes/functions.php:1336)

wp(/var/www/html/wp-blog-header.php:16)

require(/var/www/html/index.php:17)
```

**Comment by aaronjorbin** [comment]
This is odd.  It would seem as though `get_post` is returning null, even though we are inside `the_content`. I wonder if in these circumstances it would be best to just return `$the_content`?

**Comment by jeremyfelt** [comment]
@aaronjorbin good catch, thanks for reporting it! I'm _guessing_ (hoping) you're filtering the oembed response to include an excerpt. Once I did that I was able to reproduce the issue.

The oembed handler figures out what the `$post` is based on the link, and then passes that around rather than set it globally. This leads to `wp_trim_excerpt()` [not passing `$post` to its application of `the_content`](https://github.com/WordPress/wordpress-develop/blob/c01bacad6bb91c3301316737c82c004c690b9ad9/src/wp-includes/formatting.php#L3992).

So, I think you're right: best to just return `$the_content` when `$post` is not available. I've got a PR up for this in #45 if you want to take a look.

_Aside:_ It looks like we almost had an expectation of `$post_id` as available context with https://core.trac.wordpress.org/ticket/24330, but it was pulled out. [Thanks, Nacin](https://core.trac.wordpress.org/changeset/24598). There is a remainder in `do_trackbacks()` though... Maybe it's time to bring the idea back now that blocks are flying all around.

**Comment by aaronjorbin** [comment]
Yes, it looks like inside Yoast, [the oembed data](https://github.com/Yoast/wordpress-seo/blob/trunk/src/integrations/front-end/open-graph-oembed.php#L61) is filtered. 

#45 is essentially what I did to get the error out of my logs, so I approved it.  Thank you for being on top of this so quickly.

I do like the idea of bringing https://core.trac.wordpress.org/ticket/24330 back up for discussion

**Comment by jeremyfelt** [comment]
Shipped in #45 and released in 1.6.2. Thanks again! 🍺
